service/src/main/java/at/jku/isse/ecco/service/EccoService.java

```diff
@@ -53,10 +53,23 @@
 /**
  * A service class that gives access to high level operations like init, fork, pull, push, etc.
  */
+@Service
 public class EccoService implements ProgressInputStream.ProgressListener, ProgressOutputStream.ProgressListener, ReadListener, WriteListener, Closeable {
 
     private static final Logger LOGGER = Logger.getLogger(EccoService.class.getName());
 
+    public EccoService(final DispatchReader reader) {
+        this.reader = reader;
+    }
+
+    public EccoService(final DispatchWriter writer, final EntityFactory entityFactory, final TransactionStrategy transactionStrategy, final RepositoryDao repositoryDao, final RemoteDao remoteDao) {
+        this.writer = writer;
+        this.entityFactory = entityFactory;
+        this.transactionStrategy = transactionStrategy;
+        this.repositoryDao = repositoryDao;
+        this.remoteDao = remoteDao;
+    }
```

Comment:
Why are these constructors needed now? If anything calls new EccoService(null) now please add a default constructor that wraps that.